File: Data Structure/Linked List/LET_no1669_Merge In Between Linked Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        pre1, post1 = None, list1
        for i in range(b):
            if i == a - 1:
                pre1 = post1
            post1 = post1.next
        pre1.next = list2
        while list2.next:
            list2 = list2.next
        list2.next = post1.next
        post1.next = None
        
        return list1

File: Data Structure/Linked List/Medium/LET_no1669_Merge In Between Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode *prev1 = nullptr, *prev2 = list1;
        for (int i = 0; i < b; i++) {
            if (i == a - 1)
                prev1 = prev2;
            prev2 = prev2->next;
        }

        prev1->next = list2;
        while (list2->next)
            list2 = list2->next;
        list2->next = prev2->next;

        return list1;
    }
};